src/newrelic/infinite_tracing/trace_observer.go

```diff
@@ -358,36 +363,66 @@ func (to *TraceObserver) handleSupportability() {
 			// alive when a shutdown was triggered from the sender.
 			return
 		case inc := <-to.supportability.increment:
-			metrics[inc.name] += inc.count
+			v, ok := metrics[inc.name]
+			if !ok {
+				v = newEmptyMetric()
+			}
+			v[0] += inc.count
+			metrics[inc.name] = v
 		case batchCount := <-to.supportability.incrementSent:
+			v, ok := metrics[supportabilitySent]
+			if !ok {
+				v = newEmptyMetric()
+			}
 			// Since we're sending span batches, we increment by the number of spans in the batch.
-			metrics[supportabilitySent] += batchCount
+			v[0] += batchCount
+			metrics[supportabilitySent] = v
+		case dataUsage := <-to.supportability.dataUsage:
+			v, ok := metrics[supportabilityDataUsage]
+			if !ok {
+				v = newEmptyMetric()
+			}
+			v[0] += 1
+			v[1] += dataUsage
+			// There is never a body returned so no need to increment v[2]
+			metrics[supportabilityDataUsage] = v
 		case to.supportability.dump <- metrics:
 			// reset the metrics map
 			metrics = newSupportabilityMetrics()
 		}
 	}
 }
 
-func newSupportabilityMetrics() map[string]float64 {
-	// grpc codes, plus 1 for sent, and 1 for response errs
-	metrics := make(map[string]float64, numCodes+2)
+func newEmptyMetric() [6]float64 {
+	// [0] count
+	// [1] bytes sent
+	// [2] bytes received
+	// [3:] unused
+	return [6]float64{0.0, 0.0, 0.0, 0.0, 0.0, 0.0}
```

Comment:
can we abstract out `6` to a constant value here and reference that instead?

Reply:
Potentially, though the usage of [6]float for metrics (and other hard coded assumptions about the structure of metrics) is prevalent throughout all of our codebase. Unless the protocol changes the structure that NR expects for metrics, this will never change. I don't believe that setting the 6 to a constant would necessarily help in that regard.